docs/user-guide/v3_migration.rst

```diff
@@ -0,0 +1,204 @@
+   instead of directly constructing the :class:`zarr.Group` class.
+2. Deprecated most of the h5py compatibility methods. The following migration is suggested:
+
+   - Use :func:`zarr.Group.create_array` in place of :func:`zarr.Group.create_dataset`
+   - Use :func:`zarr.Group.require_array` in place of :func:`zarr.Group.require_dataset`
+
+The Store class
+~~~~~~~~~~~~~~~
+
+Some of the biggest changes in Zarr-Python 3 are found in the ``Store`` class. The most notable changes to the Store API are:
+
+1. Replaced the ``MutableMapping`` base class in favor of a custom abstract base class (:class:`zarr.abc.store.Store`).
+2. Switched to a primarily Async interface.
+
+Beyond the changes store interface, a number of deprecated stores were also removed in Zarr-Python 3:
+
+- ``N5Store``
+- ``DBMStore``
+- ``LMDBStore``
+- ``SQLiteStore``
+- ``MongoDBStore``
+- ``RedisStore``
+- ``ABSStore``
+
+Dependencies Changes
+~~~~~~~~~~~~~~~~~~~~
+
+- The new ``remote`` dependency group can be used to install a supported version of
+  ``fsspec``, required for remote data access.
+- The new ``gpu`` dependency group can be used to install a supported version of
+  ``cuda``, required for GPU functionality.
+- The ``jupyter`` optional dependency group has been removed, since v3 contains no
+  jupyter specific functionality.
+
+Configuration
+~~~~~~~~~~~~~
+
+There is a new configuration system based on `donfig <https://github.com/pytroll/donfig>`_,
+which can be accessed via :mod:`zarr.core.config`.
+Configuration values can be set using code like the following:
+
+.. code-block:: python
+
+   import zarr
+   zarr.config.set({"array.order": "F"})
+
+Alternatively, configuration values can be set using environment variables,
+e.g. ``ZARR_ARRAY__ORDER=F``.
+
+Configuration options include the following:
+
+- Default Zarr format ``default_zarr_version``
+- Default array order in memory ``array.order``
+- Default codecs ``array.v3_default_codecs`` and ``array.v2_default_compressor``
+- Whether empty chunks are written to storage ``array.write_empty_chunks``
+- Async and threading options, e.g. ``async.concurrency`` and ``threading.max_workers``
+- Selections of implementations of codecs, codec pipelines and buffers
+
+Miscellaneous
+~~~~~~~~~~~~~
+
+- The keyword argument ``zarr_version`` has been deprecated in favor of ``zarr_format``.
+
+🚧 Work in Progress 🚧
+~~~~~~~~~~~~~~~~~~~~~~
+
+Zarr-Python 3 is still under active development, and is not yet fully complete.
+The following list summarizes areas of the codebase that we expect to build out
+after the 3.0 release:
+
+- The following functions / methods have not been ported to Zarr-Python 3 yet:
+
+  * :func:`zarr.copy`
+  * :func:`zarr.copy_all`
+  * :func:`zarr.copy_store`
+  * :func:`zarr.Group.move`
+
+- The following options in the top-level API have not been ported to Zarr-Python 3 yet.
+  If these options are important to you, please open a
```

Comment:
Perhaps move this request for issues up to the top of this section, so it applies to all the bullet points and not just this one?